cronjob_update_build_run.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+repo="https://github.com/FelixStegmeier/wicked_migration_server.git"
+server_dir="./wicked_migration_server"
+migration_dir="$server_dir/wicked_migration_server"
+
+git_has_changes() {
+        main=$(git -C $migration_dir rev-parse main)
+        echo comparing hashes
+        echo "main              " $main
+        origin_main=$(git -C $migration_dir rev-parse origin/main)
+        echo "origin/main       " $origin_main
+        [ $main != $origin_main ]
+}
+
+start_server_if_not_running() {
+        pid=$(pidof wicked_migration_server)
+        if [ -z "$pid" ]; then
+                echo "starting server"
+                "$server_dir"/release/wicked_migration_server -d $server_dir/db.db3 -s $server_dir/static -i :: -p 80
+        fi
+}
+
+restart_server() {
+        pid=$(pidof wicked_migration_server)
+        if [ -n "$pid" ]; then
+                echo "stopping server with pid $pid"
+                kill "$pid"
+        fi
+        start_server_if_not_running
+}
+
+build() {
+        cargo build --release --manifest-path $migration_dir/Cargo.toml --target-dir $server_dir
+        cp -r $migration_dir/static $server_dir/static
+}
+
+echo fetching from remote $repo in $migration_dir
+git -C $migration_dir fetch origin
```

Comment:
This should probably move to the `git_has_changes` function since first AFAICT this will fail when the script gets executed for the first time and the git dir isn't cloned yet and second because it should be ensured that this is run when you parse the origin/main commit id.